addons/godot-next/resources/behavior.gd

```diff
@@ -0,0 +1,60 @@
+# author: xdgamestudios
+# license: MIT
+# description: This is an abstract base "Behavior" class for use in the "Behaviors" node class.
+#              "Behaviors" manages "Behavior" resources and calls notification methods that the Behavior implements.
+# usage: 
+# - Supported notifications:
+#     _ready() -> void
+#     _process(p_delta: float) -> void
+#     _physics_process(delta: float) -> void:
+#     _input(event: InputEvent) -> void:
+#     _unhandled_input(event: InputEvent) -> void:
+#     _unhandled_key_input(event: InputEventKey) -> void:
+#     Note:
+#     - Notifications, if present, are automatically triggered by the owner class.
+#     - If the behavior is disabled its notifications will not be processed. 
+# - Enabled Property:
+#     Individual Behavior resources can be toggled on and off using the `enabled` property.
+# - Owner Property
+#     `owner` is a reference to the Behaviors node.
+tool
+extends Resource
+class_name Behavior
+
+##### CLASSES #####
+
+##### SIGNALS #####
+
+##### CONSTANTS #####
+
+##### EXPORTS #####
+
+##### PROPERTIES #####
+
+var owner = null
+#warning-ignore:unused_class_variable
+var enabled: bool = false
```

Comment:
If you got with the other suggested changes, you will need to make this have a setter. If so, you might as well also give it a getter, change the name to have a leading underscore, and that will also hide any warnings associated with it (so you'll be able to remove the #"warning-ignore" bit as an added bonus. Then you can put the property comments above the properties.